script.js

```diff
@@ -1,147 +1,146 @@
-    postComment();
- 
-}})
-
- // обработчик клика при нажатии на кнопку "написать"
-
-buttonComment.addEventListener("click", () => {
-
-  if (textInputElement.value === '' || textareaInputElement.value === '') {
-    if (textInputElement.value === '') 
-    textInputElement.classList.add('error');
-    if (textareaInputElement.value === '')
-      textareaInputElement.classList.add('error');
-    return;
-}
- postComment()
-
- textInputElement.classList.remove('error');
- textareaInputElement.classList.remove('error');
-
- renderComments();
-}) 
-
-// обработчик клика при нажатии на кнопку "удалить последний комментарий"
-
-removeComment.addEventListener("click", () => {
-    listElement.removeChild(listElement.lastElementChild);
-});
-
-
-answer();
-initAddLike();
-renderComments();
-
-
-console.log("It works!");
\ No newline at end of file
+import { renderLoginComponent } from './login-component.js';
+import { addComment, getComments, deleteComment} from './api.js';
+import { getListComments, initLikeListeners, initCommentListeners } from './export_function.js';
+import { siteView } from './siteView.js';
+
+
+const appEl = document.getElementById("app");
+let listComments = [];
+let token =null;
+let user = null;
+
+const updateComments = (firstLoading = false) => {
+
+  // Первая загрузка
+  if (firstLoading) {
+    renderComments(listComments, true);
+  }
+  // Чтение данных по API и обновление экрана
+  getComments(token).then((responseData) => {
+    const listComments= responseData.comments.map((comment) => getListComments(comment));
+
+
+    renderComments(listComments);
+  });
+};
+
+function postComment(TextAreaElement, InputElement) {
+
+  const name = InputElement.value.replaceAll("<", "&lt;").replaceAll(">", "&gt;");
+  const text = TextAreaElement.value.replaceAll("<", "&lt;").replaceAll(">", "&gt;");
+
+  addComment(token,name, text).then(() => {
+    TextAreaElement.value = "";
+    InputElement.value = "";
+    updateComments();
+  }).catch((error) => {
+    if (error.message === 'Сервер не отвечает') {
+      console.log('500');
+      postComment(TextAreaElement, InputElement);
+    }
+    const oldInputElement = InputElement.value;
+    const oldTextAreaElement = TextAreaElement.value;
+
+    renderComments(listComments);
+
+    InputElement.value = oldInputElement;
+    TextAreaElement.value = oldTextAreaElement;
+  })
+}
+
+
+const deleteButtonListener = (listComments) => {
+  // Завешиваем обработчик кнопки "Удалить последний комментарий"
+  const deleteButtonElement = document.getElementById("remove-comment");
+  deleteButtonElement.addEventListener("click", () => {
+    if (!confirm("Вы действительно хотите удалить последний комментарий?")) {
+      return;
+    }
+    const id=listComments[listComments.length-1].id;
+    deleteComment(token, id).then((responseData) => {
+      updateComments();
+    }).catch((error) => {
+      // if (error.message === 'Сервер не отвечает') {

```

Comment:
❗ Пустые блоки catch лучше не оставлять, всегда нужно обработать ошибку, хотябы вывести в консоль
```suggestion
console.error(error);
```

Сейчас все ошибки "подавляются", и если в коде цепочки промисов что-то сломается, ты рискуешь потратить много часов дебага, из-за того что никак ошибки не увидишь